### PR TITLE
Remove hard dependency on metadatastore

### PR DESCRIPTION
**Pull request by danielballan**
See commits msgs for details.

### CODE REVIEW

File: databroker/core.py [hunk 1]
```diff
@@ -442,6 +441,7 @@ def process(mds, fs, headers, func, fields=None, fill=False):
 
 def register_builtin_handlers(fs):
     "Register all the handlers built in to filestore."
+    from filestore import handlers, HandlerBase
```
Comment (tacaswell) [inline databroker/core.py]: This is a compelling case that handlers should live in the databroker

File: databroker/broker.py [hunk 2]
```diff
@@ -19,7 +19,103 @@
                    get_fields,  # for conveniece
                    ALL
                   )
-from metadatastore.core import _format_time
+
+
+def _format_time(search_dict, tz):
```
Comment (tacaswell) [inline databroker/broker.py]: Is this functionality going to move up to databroker as well?

File: databroker/core.py (hunk 1 above)
Comment (arkilic) [inline databroker/core.py]: 👍

File: databroker/broker.py (hunk 2 above)
Comment (danielballan) [inline databroker/broker.py]: It already has.

File: databroker/core.py (hunk 1 above)
Comment (danielballan) [inline databroker/core.py]: Yeah, I think that's right. Let's keep that in mind but wait until we are sure how the packaging in going to settle out overall.

**Comment by danielballan** [comment]
The only test still failing is the portable-mds threading-related one. Ready to merge.

**Comment by tacaswell** [comment]
Can you put an `pytset.xfail` on that test?

**Comment by danielballan** [comment]
Done. Looks like we have :+1: all around, so I'm merging this to fix Travis builds for other PRs.

**Pull request merged**